hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -944,10 +953,23 @@ public void startCommitWithTime(String instantTime) {
   }
 
   /**
-   * Completes a new commit time for a write operation (insert/update/delete/insert_overwrite/insert_overwrite_table) with specified action.
+   * Provides a new commit at the specified time with the provided action type.
+   * This is only for testing purposes to setup particular sequences of commits.
+   * @param instantTime the commit start time
+   * @param actionType the type of commit
+   */
+  @VisibleForTesting
+  void startCommitWithTime(String instantTime, String actionType) {
+    HoodieTableMetaClient metaClient = createMetaClient(true);
+    startCommitWithTime(instantTime, actionType, metaClient);
+  }
+
+  /**
+   * Starts a new commit time for a write operation against the metadata table with the provided instant and action type.
    */
-  public void startCommitWithTime(String instantTime, String actionType) {
+  public void startCommitForMetadataTable(String instantTime, String actionType) {
     HoodieTableMetaClient metaClient = createMetaClient(true);
+    ValidationUtils.checkArgument(metaClient.isMetadataTable(), "Attempting to create an instant with a predetermined time on a non-metadata table.");
```

Comment:
nit: `metaClient.isMetadataTable()` check is based on the table base path.  So the validation can be done by `config.getBasePath()` first; and then call `startCommitWithTime(instantTime, actionType)` for reuse.

Reply:
I'm passing in the meta client now for the mdt so this looks a bit different

Reply:
Sg